sound/soc/soc-pcm.c

```diff
@@ -2789,6 +2788,114 @@ int soc_dpcm_be_digital_mute(struct snd_soc_pcm_runtime *fe, int mute)
 	return 0;
 }
 
+/*
+ * create a virtual FE DAI link
+ * Virtual FE DAI links are used in hostless pipelines
+ * to enable the codecs when the pipeline is triggered
+ */
+int soc_dpcm_vfe_new(struct snd_soc_card *card, int index,
+		     const char *link_name, const char *cpu_dai_name,
+		     const char *platform_name)
+{
+	struct snd_soc_dai_link *link;
+
+	link = kzalloc(sizeof(*link), GFP_KERNEL);
+	if (!link)
+		return -ENOMEM;
+
+	dev_dbg(card->dev, "ASoC: adding new virtual FE DAI link %s\n",
+		link_name);
+
+	/* define virtual FE DAI link */
+	link->virtual = 1;
+	link->name = link_name;
+	link->id = 1;
```

Comment:
Pass ID in ?

Reply:
I removed the ID as it isnt really needed